### DISCUSSION TITLE
Database no longer downgradable? (101 => 100)

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#15031] [Q&A] [answered]
I just got an error message I haven't seen for a long time: 

"Can't downgrade database from version 101 to 100: 101 is not downward compatible"

Happened to me in two different emulators after I switched from `master` (DB version 101) to `release` (DB version 100), although version 101 is listed as being backward compatible in `DataStore.java:DBVERSIONS_DOWNWARD_COMPATIBLE`.

Anyone else affected by this?
Any idea how that can happen?
Is this some general problem possibly affecting all users or all nightly users, or is it no general problem?

**Comment by eddiemuc** [comment]
You might have stumbled over a bug here in downgrade-compatibility-detection. If it is indeed a bug then it was undetected for a long time though...

Problem might be that the list of db-downgrade-compatible versions in db is updated on every db upgrade, but not on initial db creation.  Can you check your db extension table whether it contains an entry for downgradeable db versions (_type = 3, _key = ALL)?

**Comment by eddiemuc** [comment]
If my theory is correct then the problem should vanish as soon as you once upgrade from a db in version 100 (release) to version 101 (master). After that, the mentioned db extension row should exist and downgrade should be possible.

**Comment by moving-bits** [comment]
Thanks for the pointers, I'll look into it.

**Comment by eddiemuc** [comment, marked as answer]
@moving-bits: PR #15036 should solve the issue for the future (means: for databases freshly created after the PR was applied)
